crates/zkvm/lib/src/ecdsa/affine.rs

```diff
@@ -0,0 +1,219 @@
+            sec1::Coordinates::Compact { x } => Self::decompact(x),
+            sec1::Coordinates::Compressed { x, y_is_odd } => {
+                AffinePoint::<C>::decompress(x, Choice::from(y_is_odd as u8))
+            }
+            sec1::Coordinates::Uncompressed { x, y } => {
+                let x = FieldElement::<C>::from_bytes(x);
+                let y = FieldElement::<C>::from_bytes(y);
+
+                x.and_then(|x| {
+                    y.and_then(|y| {
+                        // Ensure the point is on the curve.
+                        let lhs = (y * y).normalize();
+                        let rhs = (x * x * x) + (C::EQUATION_A * x) + C::EQUATION_B;
+
+                        let point = Self::from_field_elements_unchecked(x, y);
+
+                        CtOption::new(point, lhs.ct_eq(&rhs.normalize()))
+                    })
+                })
+            }
+        }
+    }
+}
+
+impl<C: ECDSACurve> ToEncodedPoint<C> for AffinePoint<C> {
+    fn to_encoded_point(&self, compress: bool) -> EncodedPoint<C> {
+        // If the point is the identity point, we can just return the identity point.
+        if self.is_identity().into() {
+            return EncodedPoint::<C>::identity();
+        }
+
+        let (x, y) = self.field_elements();
+
+        EncodedPoint::<C>::from_affine_coordinates(&x.to_bytes(), &y.to_bytes(), compress)
+    }
+}
+
+impl<C: ECDSACurve> DecompressPoint<C> for AffinePoint<C> {
+    fn decompress(x_bytes: &FieldBytes<C>, y_is_odd: Choice) -> CtOption<Self> {
+        FieldElement::<C>::from_bytes(x_bytes).and_then(|x| {
+            let alpha = (x * x * x) + (C::EQUATION_A * x) + C::EQUATION_B;
+            let beta = alpha.sqrt();
+
+            beta.map(|beta| {
+                let y = FieldElement::<C>::conditional_select(
+                    &beta.neg(),
+                    &beta,
+                    beta.is_odd().ct_eq(&y_is_odd),
+                );
+
+                // X is normalized by virtue of being created via `FromBytes`.
+                AffinePoint::from_field_elements_unchecked(x, y.normalize())
+            })
+        })
+    }
+}
+
+impl<C: ECDSACurve> DecompactPoint<C> for AffinePoint<C> {
+    fn decompact(x_bytes: &FieldBytes<C>) -> CtOption<Self> {
+        Self::decompress(x_bytes, Choice::from(0))
+    }
+}
+
+impl<C: ECDSACurve> AffineCoordinates for AffinePoint<C> {
+    type FieldRepr = FieldBytes<C>;
+
+    fn x(&self) -> FieldBytes<C> {
+        let (x, _) = self.field_elements();
+
+        x.to_bytes()
+    }
+
+    fn y_is_odd(&self) -> Choice {
+        let (_, y) = self.field_elements();
+
+        // As field elements are create via `FromBytes`, they are already normalized.
+        y.is_odd()
+    }
+}
+
+impl<C: ECDSACurve> ConditionallySelectable for AffinePoint<C> {
+    fn conditional_select(a: &Self, b: &Self, choice: Choice) -> Self {
+        // In the vm, we dont care about constant time selection.
```

Comment:
Please elaborate on this comment further. What is `ConditionallySelectable` used for? 

Once you fix, make it not in first-person tense.